File: src/main/java/com/dems/evidence/EvidenceService.java
package com.dems.evidence;

import com.dems.enums.EvidenceStatus;
import com.dems.enums.EvidenceType;
import org.springframework.core.io.Resource;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.time.LocalDate;

/**
 * Service interface managing digital evidence uploads, file storage orchestration, downloads, and metadata updates.
 */
public interface EvidenceService {

    EvidenceResponse uploadEvidence(MultipartFile file, EvidenceUploadRequest request, String uploaderEmail);

    EvidenceResponse getEvidenceById(Long id);

    Page<EvidenceSummaryResponse> getAllEvidence(Pageable pageable);

    Page<EvidenceSummaryResponse> getEvidenceByCaseId(Long caseId, Pageable pageable);

    Resource downloadEvidence(Long id);

    EvidenceResponse updateEvidence(Long id, EvidenceUpdateRequest request);

    EvidenceResponse updateEvidenceStatus(Long id, EvidenceStatusUpdateRequest request);

    Page<EvidenceSummaryResponse> searchEvidence(
            EvidenceStatus status,
            EvidenceType evidenceType,
            Long caseId,
            Long uploadedById,
            LocalDate uploadDate,
            String evidenceNumber,
            String evidenceName,
            String searchTerm,
            Boolean active,
            Pageable pageable
    );
}

File: frontend/src/services/evidenceService.js
import apiClient from './apiClient';
import { API_ENDPOINTS } from '../constants/apiEndpoints';

export const evidenceService = {
  uploadEvidence: async (formData) => {
    return await apiClient.post(API_ENDPOINTS.EVIDENCE.UPLOAD, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
  },

  getAllEvidence: async (params = {}) => {
    return await apiClient.get(API_ENDPOINTS.EVIDENCE.BASE, { params });
  },

  getEvidenceById: async (id) => {
    return await apiClient.get(API_ENDPOINTS.EVIDENCE.BY_ID(id));
  },

  downloadEvidence: async (id) => {
    return await apiClient.get(API_ENDPOINTS.EVIDENCE.DOWNLOAD(id), {
      responseType: 'blob',
    });
  },

  updateEvidence: async (id, requestData) => {
    return await apiClient.put(API_ENDPOINTS.EVIDENCE.BY_ID(id), requestData);
  },

  updateEvidenceStatus: async (id, statusData) => {
    return await apiClient.patch(API_ENDPOINTS.EVIDENCE.STATUS(id), statusData);
  },

  getEvidenceByCaseId: async (caseId, params = {}) => {
    return await apiClient.get(API_ENDPOINTS.EVIDENCE.BY_CASE(caseId), { params });
  },

  searchEvidence: async (searchParams = {}) => {
    return await apiClient.get(API_ENDPOINTS.EVIDENCE.SEARCH, { params: searchParams });
  },
};

export default evidenceService;
